src/Common/DoctrineHeartbeatMiddleware/ConnectionHeartbeatHandler.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\Common\DoctrineHeartbeatMiddleware;
+
+use Gaming\Common\Scheduler\Handler;
+use Gaming\Common\Scheduler\Scheduler;
+
+final class ConnectionHeartbeatHandler implements Handler
+{
+    public function __construct(
+        private readonly TrackActivityConnection $connection,
+        private readonly string $dummySql,
+        private readonly int $interval,
+    ) {
+    }
+
+    public function handle(Scheduler $scheduler): void
+    {
+        if (
```

Comment:
If the connection is closed by `Doctrine\DBAL\Connection::close()` then the heartbeat handler will still run and keep the connection alive. There is no method for `Doctrine\DBAL\Driver\Connection` that checks whether the connection is closed or not. Either this issue can be ignored or maybe a `WeakMap` can solve it.

Reply:
`WeakReference` does the job.